src/main/java/com/mraof/minestuck/computer/editmode/EditmodeLocations.java

```diff
@@ -0,0 +1,371 @@
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.NbtOps;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.level.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.*;
+
+/**
+ * Contains a list of block positions in a radius around which an editmode player can move freely.
+ * An instance is stored in SburbConnection for the SBURB client of the connection.
+ */
+public class EditmodeLocations
+{
+	//TODO prevent going to non Medium dimensions post Entry, delete maps to the Overworld
+	
+	private static final Logger LOGGER = LogManager.getLogger();
+	
+	public static final int VALIDATION_RADIUS = 25;
+	public static final int ENTRY_RANGE = 30;
+	
+	private final Multimap<ResourceKey<Level>, Pair<BlockPos, Source>> locations = ArrayListMultimap.create();
+	
+	public enum Source
+	{
+		BLOCK,
+		ENTITY,
+		ENTRY;
+		
+		public static Source fromInt(int ordinal) //converts int back into enum
+		{
+			if(0 <= ordinal && ordinal < Source.values().length)
+				return Source.values()[ordinal];
+			else
+				throw new IllegalArgumentException("Invalid ordinal of " + ordinal + " for EditmodeLocations Source!");
+		}
+	}
+	
+	public EditmodeLocations(ResourceKey<Level> level, BlockPos pos, Source source)
+	{
+		addEntry(level, pos, source);
+	}
+	
+	public EditmodeLocations()
+	{
+	}
+	
+	public Multimap<ResourceKey<Level>, Pair<BlockPos, Source>> getLocations()
+	{
+		return locations;
+	}
+	
+	public void addEntry(ResourceKey<Level> level, BlockPos pos, Source source)
+	{
+		if(level == null || pos == null)
+			return;
+		
+		Pair<BlockPos, Source> locationPairIn = Pair.of(pos, source);
+		
+		//TODO consider validating the pos and source
+		if(locations.containsKey(level))
+		{
+			for(Pair<BlockPos, Source> locationPair : locations.get(level))
+			{
+				if(locationPair.getFirst().equals(pos) && locationPair.getSecond().equals(source))
+					return; //prevent duplicates
+			}
+		}
+		
+		locations.put(level, locationPairIn);
+	}
+	
+	public void removeEntry(ResourceKey<Level> level, BlockPos pos, Source source)
+	{
+		if(level == null || pos == null)
+			return;
+		
+		//dont try to remove ENTRY Sources as they are only added once
+		if(source != Source.ENTRY)
+			locations.remove(level, Pair.of(pos, source));
+	}
+	
+	/**
+	 * Takes in a player and looks through each source (such as a computer) providing editmode to see if the player resides within one of the spaces.
+	 * Used both server side and client side
+	 *
+	 * @param editPlayer Player in editmode
+	 * @return Whether the player is standing in a supported region
+	 */
+	public boolean canMoveAtPosition(Player editPlayer, double defaultRange)
```

Comment:
This description might be misleading as the function doesn't just check if the player can move in their position, but might also change their position or change their velocity.